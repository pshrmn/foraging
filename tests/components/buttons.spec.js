import React from 'react';
import renderer from 'react-test-renderer';
import {
  PosButton,
  NegButton,
  NeutralButton
} from 'components/common/Buttons';

describe('buttons', () => {
  describe('PosButton', () => {
    it('creates a <button> element', () => {
      const Button = renderer.create(
        <PosButton>Positive</PosButton>
      );
      const tree = Button.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  
  describe('NegButton', () => {
    it('creates a <button> element', () => {
      const Button = renderer.create(
        <NegButton>Positive</NegButton>
      );
      const tree = Button.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('NeutralButton', () => {
    it('creates a <button> element', () => {
      const Button = renderer.create(
        <NeutralButton>Positive</NeutralButton>
      );
      const tree = Button.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('sets the title attribute', () => {
      const title = 'foo';
      const Button = renderer.create(
        <NeutralButton title={title}>Positive</NeutralButton>
      );
      const tree = Button.toJSON();
      expect(tree.props.title).toBe(title)
      expect(tree).toMatchSnapshot();
    });

    it('sets the disabled attribute', () => {
      const Button = renderer.create(
        <NeutralButton disabled>Positive</NeutralButton>
      );
      const tree = Button.toJSON();
      expect(tree.props.disabled).toBe(true);
      expect(tree).toMatchSnapshot();
    });

    it('sets the button type', () => {
      const type = 'submit';
      const Button = renderer.create(
        <NeutralButton type={type}>Positive</NeutralButton>
      );
      const tree = Button.toJSON();
      expect(tree.props.type).toBe(type);
      expect(tree).toMatchSnapshot();
    });

    it('sets the onClick event using the click prop', () => {
      const Button = renderer.create(
        <NeutralButton click={() => 1}>Positive</NeutralButton>
      );
      const tree = Button.toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree.props.onClick()).toBe(1);
    });
  });
});
