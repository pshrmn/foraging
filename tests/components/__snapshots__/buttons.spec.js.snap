exports[`buttons NegButton creates a <button> element 1`] = `
<button
  className="neg "
  disabled={false}
  onClick={[Function]}
  title=""
  type="button">
  
</button>
`;

exports[`buttons NeutralButton creates a <button> element 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  title=""
  type="button">
  
</button>
`;

exports[`buttons NeutralButton sets the button type 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  title=""
  type="submit">
  
</button>
`;

exports[`buttons NeutralButton sets the disabled attribute 1`] = `
<button
  className=""
  disabled={true}
  onClick={[Function]}
  title=""
  type="button">
  
</button>
`;

exports[`buttons NeutralButton sets the onClick event using the click prop 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  title=""
  type="button">
  
</button>
`;

exports[`buttons NeutralButton sets the title attribute 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  title="foo"
  type="button">
  
</button>
`;

exports[`buttons PosButton creates a <button> element 1`] = `
<button
  className="pos "
  disabled={false}
  onClick={[Function]}
  title=""
  type="button">
  
</button>
`;
